### ISSUE TITLE
Required items

### ISSUE THREAD

**Issue by mikeher** [AssistantSMS/App#16]
Love the app. There's a minor issue with the "view all required items" it seems though. It is not normalized to single units, e.g. if you add 1x metal 3 the tree will show that 20x metal 2 is needed, then 300 metal 1, then 4500 scrap metal.

**Comment by Khaoz-Topsy** [comment]
Thank you for pointing this out! I must not have taken into account the output of the craft bot recipes and just assumed it is always 1 🤔

**Comment by Khaoz-Topsy** [comment]
![image](https://user-images.githubusercontent.com/15005470/106011928-99ca6700-60c3-11eb-8de6-5398b814cc70.png)


@mikeher Does this look more correct? I think I have fried my brains trying to get this right 😅

**Comment by mikeher** [comment]
> ![image](https://user-images.githubusercontent.com/15005470/106011928-99ca6700-60c3-11eb-8de6-5398b814cc70.png)
> 
> @mikeher Does this look more correct? I think I have fried my brains trying to get this right 😅

Wow you are fast :-) Yes that looks correct to me. And yea, all these nested dependencies are quite a headache.

In order to avoid having to evaluate the quantity you get from a single act of crafting, I would probably normalise the required amounts in the database so they indicate how much a single instance of a certain item costs, e.g. 1x Metal Block 3 costs 2x Metal Block 2, 0.2x Ember, 0.4x Water. That makes it a lot easier when dealing with seemingly infinite nesting of dependencies :P

**Comment by Khaoz-Topsy** [comment]
I am busy refactoring the whole app's code,  so thought I would attack this bug while I was busy with it 😋. I think that is an awesome idea! How would you word it? 

I am thinking that a person could choose between the two modes "Real World" or "Scientific" 🤔. And I would need something there to explain the difference 😅

**Comment by Khaoz-Topsy** [comment]
I have released the fix for the cart to the Android and iOS store. I will work on the "scientific" recipe display so long 😁

**Comment by mikeher** [comment]
Uh, that's wonderful, I just updated and tested it. Very useful! :-)

I'm not sure I know exactly what you mean, I assume it might be the fact that it can be hard to determine when to stop breaking up into smaller dependencies, e.g. most people probably won't need to break up 120 "units of water" to 120 "units of waterbucket". So real world vs scientific aims to make that determination?

End users want simplicity, so choosing a "mode" might create unwanted confusion. A suggestion could be to store a boolean, is_grindable, on every item and then let that be the stop block. E.g. water is grindable, no need to split it up into water buckets. Same goes for metal 1.

Since I am an end user I can even give you a user story :-) I found your app while searching for item requirements so I could calculate how to make a tonne of metal 3. I have a vehicle to mine a lot of metal 1 and a vehicle to pump a lot of water so I don't need to know how much scrap metal and waterbuckets I will need. I can only assume that anyone else seeking to calculate how much is needed to craft a big set of resources have also progressed far enough for their grinding to be as effective as can be.

Love the app and I will happily provide feedback on anything you deem useful.
